pandas/core/internals.py

```diff
@@ -2609,10 +2634,62 @@ class SparseBlock(NonConsolidatableMixIn, Block):
     is_sparse = True
     is_numeric = True
     _box_to_block_values = False
-    _can_hold_na = True
     _ftype = 'sparse'
     _holder = SparseArray
 
+    def __init__(self, values, placement, ndim=None, fastpath=False, **kwargs):
+        super(SparseBlock, self).__init__(values, placement,
+                                          ndim, fastpath,
+                                          **kwargs)
+
+        dtype = self.values.sp_values.dtype
+
```

Comment:
this is pretty complex, pls simplify

Reply:
I added these codes because `coerce_to_target_dtype` does not work well in SparseBlock. This is only checking the type information of SparseArray and these procedure is also implemented in `IntBlock`, etc. I cannot figure out how to simplify. Any suggestion for simplifying?

Reply:
I am not against this theorectially, but the implementation is fragile here. there are functions for validation already in the sparse classes.